src/components/login-form.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FormEventHandler, useState } from 'react';
+import PasswordInput from './password-input';
+import SubmitButton from './submit-button';
+import TextInput from './text-input';
+
+interface ILoginFormValues {
+  email?: string;
+  password?: string;
+}
+
+const LoginForm = (): React.ReactElement => {
+  const [formValues, setFormValues] = useState<ILoginFormValues>({
+    email: '',
+    password: '',
+  });
+
+  const handleSubmit: FormEventHandler = (e) => {
+    e.preventDefault();
+    const email = formValues?.email;
+    const password = formValues?.password;
+
+    if (email && password) {
+      console.log(`Logged in with email ${email} and password ${password}`);
+    }
+  };
+
+  const handleEmailChange = (emailValue: string) => {
+    setFormValues({
+      ...formValues,
+      email: emailValue,
+    });
+  };
+
+  const handlePasswordChange = (passwordValue: string) => {
+    setFormValues({
+      ...formValues,
+      password: passwordValue,
+    });
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div>
+        <TextInput
+          name='email'
+          label='E-mail'
+          value={formValues?.email}
+          onValueChange={(value) => handleEmailChange(value)}
```

Comment:
Quando colocamos uma função com parâmetros iguais aos que o componente passa, podemos simplificar passando a função direto. Então essa forma seria mais simples e produziria o mesmo resultado:
```suggestion
          onValueChange={handleEmailChange}
```